modeltranslation/utils.py

```diff
@@ -13,46 +16,48 @@
     set_enable_fallbacks,
 )
 
+from ._typing import AutoPopulate
+
+_T = TypeVar("_T")
+
 
-def get_language():
+def get_language() -> str | None:
     """
     Return an active language code that is guaranteed to be in
     settings.LANGUAGES (Django does not seem to guarantee this for us).
     """
     lang = _get_language()
-    if lang is None:  # Django >= 1.8
-        return settings.DEFAULT_LANGUAGE
```

Comment:
This has no relation to adding a types, revert it.

Reply:
Ok after this one I will make two separate PRs:
- Remove old compat code
- Use pyupgrade (via ruff) (and maybe enforce it at the CI level)